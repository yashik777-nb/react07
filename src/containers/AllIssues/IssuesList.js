import React from "react";
import Issue from "../../components/Issue/Issue";
import IssuesListCSS from "./IssuesList.module.css";

export default class IssuesList extends React.Component {
  render() {
    let issuesList = this.props.issues.map((issue) => (
      <Issue
        key={issue.id}
        id={issue.id}
        issueDescription={issue.issueDescription}
        severity={issue.severity}
        status={issue.status}
      />
    ));

    return (
      <div>
        <table className={IssuesListCSS.border}>
          <thead>
            <tr>
              <th>ID</th>
              <th>Issue Description</th>
              <th>Severity</th>
              <th>Status</th>
            </tr>
          </thead>
          <tbody>{issuesList}</tbody>
        </table>
      </div>
    );
  }
}
