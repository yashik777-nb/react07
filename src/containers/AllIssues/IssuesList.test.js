import { configure, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import IssuesList from "./IssuesList";

configure({ adapter: new Adapter() });

const issues = [
  {
    id: 1,
    issueDescription: "Issue Description One",
    severity: "Critical",
    status: "Open",
  },
  {
    id: 2,
    issueDescription: "Issue Description Two",
    severity: "Minor",
    status: "Closed",
  },
  {
    id: 3,
    issueDescription: "Issue Description Three",
    severity: "Major",
    status: "In Progress",
  },
];

describe("Issues List Page Test", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<IssuesList issues={issues} />);
  });

  it("IssuesList -  Render Issues Description as Second Column", () => {
    expect(wrapper.find("th").at(1).props().children).toEqual(
      "Issue Description"
    );
  });

  it("IssuesList -  Render Issues Description as Fourth Column", () => {
    expect(wrapper.find("th").at(3).props().children).toEqual("Status");
  });

  it("IssuesList -  Render 2nd Issue Description as Props to Second Issue Compoent", () => {
    expect(wrapper.find("Issue").at(1).props().issueDescription).toEqual(
      "Issue Description Two"
    );
  });

  it("IssuesList -  Render 2nd Issue Status as Props to Second Issue Compoent", () => {
    expect(wrapper.find("Issue").at(1).props().status).toEqual("Closed");
  });

  test("Issues List Page Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
