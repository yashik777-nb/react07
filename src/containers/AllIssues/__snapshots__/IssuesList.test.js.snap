// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issues List Page Test Issues List Page Snapshot 1`] = `ReactWrapper {}`;
